### PR TITLE
Upgrade data table

### PR DESCRIPTION
**Pull request by ronnietaylor**
Still in progress....

### CODE REVIEW

File: lib/data-table/table.rb [hunk 1]
```diff
@@ -119,7 +118,7 @@ def render_rows(collection)
       html = ""
       collection.each_with_index do |row, row_index|
         css_class = @alternate_rows && row_index % 2 == 1 ? 'alt ' : ''
-        if @row_style && style = @row_style.call(row, row_index)
+        if @row_style && style == @row_style.call(row, row_index)
```
Comment (aharpervc) [inline lib/data-table/table.rb]: This may actually need to be assignment, otherwise what is `style`

Comment (ronnietaylor) [inline lib/data-table/table.rb]: Good point and thanks. Fixed.

**Comment by aharpervc** [comment]
A working CI configuration, be still my heart.

File: spec/enum_spec.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+require 'spec_helper'
+
+describe Enumerable do
+  context "with a non-empty collection of hashes" do
+    let(:collection) {
+      [
+        {name: 'Luke Skywalker', class: 'Jedi Knight', world: 'Star Wars', power_level: 50},
+        {name: 'Emporer Palpatine', class: 'Sith Lord', world: 'Star Wars', power_level: 95},
+        {name: 'Mithrander', class: 'Wizard', world: 'Middle Earth', power_level: 9001},
```
Comment (aharpervc) [inline spec/enum_spec.rb:9]: Mithrandir

File: spec/enum_spec.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
+require 'spec_helper'
+
+describe Enumerable do
+  context "with a non-empty collection of hashes" do
+    let(:collection) {
+      [
+        {name: 'Luke Skywalker', class: 'Jedi Knight', world: 'Star Wars', power_level: 50},
+        {name: 'Emporer Palpatine', class: 'Sith Lord', world: 'Star Wars', power_level: 95},
```
Comment (aharpervc) [inline spec/enum_spec.rb:8]: Emperor

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -1,12 +1,12 @@
 require 'data-table'
+require 'pry'
```
Comment (aharpervc) [inline spec/spec_helper.rb:2]: `pry` isn't in the gemspec

File: examples/all_features.rb [hunk 5]
```diff
@@ -0,0 +1,161 @@
+require 'byebug'
+require 'pry'
+require 'pry-byebug'
```
Comment (aharpervc) [inline examples/all_features.rb:3]: All three of these are missing from the gemspec

**Comment by baueric** [comment]
Preparation for locking apps at data-table v1.0 and moving axiom-rails to v2.0

[Branch for version 1](https://github.com/veracross/data-table/tree/v1)

Apps that include data-table gem:

- [x] [axiom-rails](https://github.com/axiom-dev/axiom-rails/blob/f057aababdc6425e0d2cdd4ce7245c4075b0861e/axiom-rails.gemspec#L23) - already fixed to 1.0
- [x] [syringe](https://github.com/veracross/syringe/blob/7f8cdcbb79e349241129b7c406c23b67fe67fcec/Gemfile#L11) [commit](https://github.com/veracross/syringe/commit/b1d2a9d63723fc7d48ea04587129dc07632cc16f)
- [x] [job_calendar](https://github.com/veracross/job_calendar/blob/8cc961d7ea8a82d2832e2a9e36a10260d95b73b5/Gemfile#L10) - [commit](https://github.com/veracross/job_calendar/commit/06a520bcf8e692b2be88d5868b49d9ba1542c5c3)
- [x] [portals](https://github.com/veracross/portals/blob/23320b70127f1f7b08df9dfafea5e3103f9338f8/Gemfile#L23) [commit](https://github.com/veracross/portals/commit/25f9a5a0bb68e0fcd96e2f0d89428636dfbb244a)
- [x] [wiki](https://github.com/veracross/wiki/blob/2009825c970a240b2f2549cc2d81a063f096df3d/Gemfile#L7) [commit](https://github.com/veracross/wiki/commit/f0e20ed63618ac457dcf425d3517df3fadaf4bea)
- [x] [utilities](https://github.com/veracross/utilities/blob/821ade20e8692e274286cb73f14c485c9e307121/Gemfile#L10) [commit](https://github.com/veracross/utilities/commit/82fab92d6e51b7c1d3f88efc35bc5e7470ce88e9)
- [x] [importer](https://github.com/veracross/importer/blob/d07bc1a77e2513575ffc4f5d877e63838a473bde/Gemfile#L49) - [commit](https://github.com/veracross/importer/commit/d69bd93861ad70a705a79865cdcc016227e42eff)
- [x] [sunflower](https://github.com/veracross/sunflower/blob/735cf9d0b3be56d834e617bc435f147e4da96f7f/Gemfile#L33) - [commit](https://github.com/veracross/sunflower/commit/475a1cf58b1e67523a80593edc7cd4f6cce2e4a8)
- [x] [sherpa](https://github.com/veracross/sherpa/blob/0a4fc509cc894ee00697652de190da24706e2c8d/Gemfile#L25) - [commit](https://github.com/veracross/sherpa/commit/cfac8cef66a38d60df69e31966014e324b472e7c)
- [x] [reckoner](https://github.com/veracross/reckoner/blob/228280c5934a819f9dd0b46517c43bcd46c223a8/Gemfile#L22) - [commit](https://github.com/veracross/reckoner/commit/0ac2606994fccca5eec20270488cf7817d0ed516)

**Pull request merged**